R/MultipleTables-method-exported.R

```diff
@@ -0,0 +1,165 @@
+                                   method = 'exact',
+                                   verbose = FALSE, 
+                                   control =  list()){
+  
+  ## check input
+  if (!inherits(multiple_tables_object, "MultipleTables"))
+    stop("Use only with 'MultipleTables' objects.\n")
+  
+  multiple_tables_object$method <- checkMethodArgument(method, 
+                                        multiple_tables_object$measure, 
+                                        verbose = verbose)
+  
+  
+  
+  ### load control list
+  control <- MultipleTables._setControlList(multiple_tables_object, control)
+  
+  
+  
+  ## Estimate prior parameters and do likelihood ratio test for correlation
+  prior_estimate_and_test <- MultipleTables._priorParemetersEstimate(multiple_tables_object, control, verbose)
+  multiple_tables_object$chi2_value <- prior_estimate_and_test$chi2_value
+  multiple_tables_object$p_value <- prior_estimate_and_test$p_value
+  multiple_tables_object$prior_mle <- prior_estimate_and_test$prior_mle
+  multiple_tables_object$cov_matrix_log <- prior_estimate_and_test$cov_matrix_log
+  multiple_tables_object$hessian_log <- prior_estimate_and_test$hessian_log
+  
+  
+
+  ## draw posterior samples 
+  multiple_tables_object$samples <- MultipleTables._studySpecificPosteriorSamples(multiple_tables_object, control)
+  
+  # calculate density  for study specific measures
+  multiple_tables_object$density <- MultipleTables._studySpecificPosteriorDensity(multiple_tables_object, control)
+  
+  
+  invisible(multiple_tables_object)
+  
+}
+
+
+MultipleTables.summary<- function(multiple_tables_object, 
+                                  alpha = 0.05,
+                                  verbose = TRUE, 
+                                  digit = 3,
+                                  control = list() ){
+  ## check input
+  checkAlpha(alpha)
+  checkDigit(digit)
+  MultipleTables._isObjectFitted(multiple_tables_object)
+  
+  ## update object
+  multiple_tables_object$alpha <- alpha
+  multiple_tables_object$digit <- digit
+  
+  ### load control list
+  control <- MultipleTables._setControlList(multiple_tables_object, control)
+  
+  
+  ## Estimate overall measure(point and Wald confident internal)
+  overall_measure_estimate <- MultipleTables._overallMeasureEstimate(multiple_tables_object)
+  overall_measure_estimation <- list(point = overall_measure_estimate$overall,
+                                    confindent_interval = overall_measure_estimate$CI)
+  multiple_tables_object$overall_measure_estimation <- overall_measure_estimation
+  
+  
+  ## Estimate study specific summary
+  multiple_tables_object$specific_summary <- MultipleTables._StudySpecificMeasureEstimate(
+                                              multiple_tables_object, 
+                                              control)
+  
+  ## print out report if user requests
+  if(verbose){
+    MultipleTables._summaryPrint(multiple_tables_object)
```

Comment:
Only when verbose == True, the MultipleTables._summaryPrint(multiple_tables_object) function will be called then the cat functions within MultipleTables._summaryPrint() can be called.